### PR TITLE
Feature/issue 186 create hal rest endpoint

### PR DESCRIPTION
**Pull request by mbarber-gala**
Created Hal Rest Controller and Service. Created tests for both classes.

### CODE REVIEW

File: src/main/java/org/galatea/starter/entrypoint/HalRestController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  /**
+   * Send the received text to the HalService to be processed and send the result out
+   */
+  // @GetMapping to link http GET request to this method
+  // @RequestParam to take a parameter from the url
+  @GetMapping(value = "${webservice.halpath}", produces = {MediaType.APPLICATION_JSON_VALUE})
+  public String halEndpoint(@RequestParam(value = "text", required = true) String text)
+      throws JsonProcessingException {
+    ObjectMapper mapper = new ObjectMapper();
```
Comment (paulo-reichert) [inline src/main/java/org/galatea/starter/entrypoint/HalRestController.java:41]: You shouldn't need to manually use the ObjectMapper here.

File: src/main/java/org/galatea/starter/service/HalService.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import net.sf.aspect4log.Log;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@ToString
+@EqualsAndHashCode
+@Slf4j
+@Log
+@Service
+public class HalService {
```
Comment (paulo-reichert) [inline src/main/java/org/galatea/starter/service/HalService.java:17]: for now we can just have empty methods that either return an empty string or nothing. as we implement the real functionality we will fill those methods.

File: src/test/java/org/galatea/starter/entrypoint/HalRestControllerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+@ToString
+@EqualsAndHashCode
+@Category(org.galatea.starter.IntegrationTestCategory.class)
+public class HalRestControllerIntegrationTest {
+
+  @Test
+  public void testCoinFlip() {
+    String fuseHostName = System.getProperty("fuse.sandbox.url");
+    if (fuseHostName == null || fuseHostName.isEmpty()) {
+      // TODO: the base URL should probably be moved to a src/test/resources properties file
```
Comment (paulo-reichert) [inline src/test/java/org/galatea/starter/entrypoint/HalRestControllerIntegrationTest.java]: Sonar will complain about the new TODO here. TODOs are flagged as code debt and we have rules against adding those.

File: src/main/java/org/galatea/starter/service/HalService.java (hunk 2 above)
Comment (mbarber-gala) [reply src/main/java/org/galatea/starter/service/HalService.java:17]: - [x] Return empty strings for methods without final implementation

File: src/test/java/org/galatea/starter/entrypoint/HalRestControllerIntegrationTest.java (hunk 3 above)
Comment (mbarber-gala) [reply src/test/java/org/galatea/starter/entrypoint/HalRestControllerIntegrationTest.java]: - [x] Remove todos
- [x] Move base URL to a resource file

File: src/main/java/org/galatea/starter/entrypoint/HalRestController.java (hunk 1 above)
Comment (mbarber-gala) [reply src/main/java/org/galatea/starter/entrypoint/HalRestController.java]: - [ ] Remove need to use ObjectMapper manually

Comment (mbarber-gala) [reply src/main/java/org/galatea/starter/entrypoint/HalRestController.java:41]: Without the manual call to the object mapper the returned string doesn't get converted into JSON. Looking at POSTMAN I get "Unexpected Token 'T'" for the response

**Review by paulo-reichert — APPROVED** [review]
Good to go.

**Pull request merged**